### PR TITLE
pnpm with lerna update

### PR DESCRIPTION
**Pull request by AndrejNikolicEq**
# pnpm with lerna 7.2 update

### Type
- Maintenance (deps only)

### CODE REVIEW

**Comment by AndrejNikolicEq** [comment]
still WIP, added for testing pnpm in workflow

**Review by eikeland — APPROVED** [review]

**Comment by AndrejNikolicEq** [comment]
Complete

**Comment by AndrejNikolicEq** [comment]
added changeset

File: package.json [hunk 1]
```diff
@@ -4,22 +4,21 @@
   "description": "collection of fusion components",
   "main": "index.js",
   "scripts": {
-    "start": "npm run storybook",
-    "install": "npm run bootstrap",
-    "bootstrap": "lerna bootstrap -- --legacy-peer-deps",
+    "start": "pnpm run storybook",
+    "clean": "rm -rf node_modules && rm -rf pnpm-lock.yaml && lerna clean -y",
```
Comment (odinr) [inline package.json:8]: this should be using rimraf so it will be MS comp

File: .github/workflows/chromatic.yml [hunk 2]
```diff
@@ -24,12 +24,15 @@ jobs:
       - uses: actions/setup-node@v2
         with:
           node-version: '16'
-          cache: 'npm'
+
+      - name: Install pnpm
+        run: npm install pnpm -g
+
```
Comment (odinr) [inline .github/workflows/chromatic.yml:30]: nope!
use the correct action, which should be executed before npm install, for caching pnpm modules

File: .github/workflows/generate-docs.yml [hunk 3]
```diff
@@ -32,19 +32,21 @@ jobs:
         uses: actions/setup-node@v2
         with:
           node-version: '16'
-          cache: 'npm'
+
+      - name: Install pnpm
+        run: npm install pnpm -g
```
Comment (odinr) [inline .github/workflows/generate-docs.yml:37]: see previous commnent about pnpm init

File: .github/workflows/lint.yml [hunk 4]
```diff
@@ -18,19 +18,19 @@ jobs:
         uses: actions/setup-node@v2
         with:
           node-version: '16'
-          cache: 'npm'
+
+      - name: Install pnpm
+        run: npm install pnpm -g
```
Comment (odinr) [inline .github/workflows/lint.yml:23]: see above

File: .github/workflows/pre-release.yml [hunk 5]
```diff
@@ -38,27 +36,32 @@ jobs:
           path: node_modules
           key: ${{ runner.OS }}-build-${{ hashFiles('**/package.json') }}
 
+      - name: Install pnpm
+        run: npm install pnpm -g
```
Comment (odinr) [inline .github/workflows/pre-release.yml:40]: see above

File: .github/workflows/pre-release.yml [hunk 6]
```diff
@@ -38,27 +36,32 @@ jobs:
...
-      - name: Release packages
-        run: npx lerna publish from-package --yes --no-verify-access --dist-tag ${{ github.event.inputs.tag }}
+      - name: Publish versions (pre-release)
+        run: |
+          pnpm changeset publish
+          git push --follow-tags
         env:
-          NODE_AUTH_TOKEN: ${{ secrets.NPM_AUTH_TOKEN }}
\ No newline at end of file
+          GH_TOKEN: ${{ github.token }}
```
Comment (odinr) [inline .github/workflows/pre-release.yml:67]: Nope, read docs

**Review by odinr — CHANGES_REQUESTED** [review]
we need to talk before this is merged

File: .github/workflows/pre-release.yml (hunk 6 above)
Comment (AndrejNikolicEq) [reply .github/workflows/pre-release.yml]: where? looks ok here https://github.com/changesets/changesets/blob/main/docs/prereleases.md

File: .github/workflows/pre-release.yml [hunk 7]
```diff
@@ -18,47 +18,34 @@ jobs:
...
-      - name: Install Node Dependencies
-        if: steps.cache-node.outputs.cache-hit != 'true'
-        run: npm ci --legacy-peer-deps
-        env:
-          CI: TRUE
-
       - name: Bump versions (pre-release)
-        run: npx lerna version --yes --conventional-prerelease --no-changelog --no-git-tag-version --preid ${{ github.event.inputs.tag }}
+        run: |
+          pnpm changeset pre enter next
```
Comment (odinr) [inline .github/workflows/pre-release.yml:32]: read about [pre release](https://github.com/changesets/changesets/blob/main/docs/prereleases.md)
this is to advance to use within a pipeline, use [snapshot](https://github.com/changesets/changesets/blob/main/docs/snapshot-releases.md)

File: .github/workflows/pre-release.yml [hunk 8]
```diff
@@ -18,47 +18,34 @@ jobs:
...
+          pnpm changeset version
         env:
           GH_TOKEN: ${{ github.token }}
 
       - name: Commit changes
         continue-on-error: true
         run: |
             git add packages/*/package.json storybook/package.json 
-            git commit -m'chore(prerelease): bumped version with tag ${{ github.event.inputs.tag }}'
+            git commit -m'chore(prerelease): enter prerelease mode and version packages with tag ${{ github.event.inputs.tag }}'
```
Comment (odinr) [inline .github/workflows/pre-release.yml:41]: this wont work

File: .github/workflows/pre-release.yml [hunk 9]
```diff
@@ -18,47 +18,34 @@ jobs:
...
             git push
         env:
           GH_TOKEN: ${{ github.token }}
 
-      - name: Release packages
-        run: npx lerna publish from-package --yes --no-verify-access --dist-tag ${{ github.event.inputs.tag }}
+      - name: Publish versions (pre-release)
+        run: |
```
Comment (odinr) [inline .github/workflows/pre-release.yml:47]: no, read the documentation, we do NOT want to release a pre-release as latests

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -17,21 +17,20 @@ description
 
 # Setup
 
-## Bootstrap Repo
+## Install
 
 ```sh
-npm install
+npm install pnpm -g
```
Comment (odinr) [inline CONTRIBUTING.md:23]: should reference to PNPM doc side

File: packages/filter/src/FilterProvider.tsx [hunk 11]
```diff
@@ -18,45 +18,47 @@ export type FilterProviderProps<TSelections extends Record<string, unknown>, TDa
   initialFilters?: Record<string, Filter<TData, TSelections>>;
 };
 
-export const FilterProvider = <
+export const FilterProvider: <
```
Comment (odinr) [inline packages/filter/src/FilterProvider.tsx:21]: why is this file modified in this PR

File: packages/filter/src/components/filter/checkbox/CheckboxFilterProvider.tsx [hunk 12]
```diff
@@ -9,9 +9,9 @@ import type { FilterFn } from '../../../types';
 
 const createFilterFn =
   <TData extends Record<string, unknown>, TValue = string>(selector: FilterOptionSelector<TData>) =>
-  (data: TData[], selection: Set<TValue>) => {
-    return selection.size ? data.filter((x) => selection.has(selector(x).key as unknown as TValue)) : data;
-  };
+    (data: TData[], selection: Set<TValue>) => {
```
Comment (odinr) [inline packages/filter/src/components/filter/checkbox/CheckboxFilterProvider.tsx:12]: why is this file modified in this PR

File: packages/filter/src/components/panel/FilterPanel.tsx [hunk 13]
```diff
@@ -72,7 +72,9 @@ export type FilterPanelProps<TData> = JSX.IntrinsicElements['div'] & {
 /**
  * Base component for displaying filter components and controllers
  */
-export const FilterPanel = <TData,>(props: React.PropsWithChildren<FilterPanelProps<TData>>): JSX.Element => {
+export const FilterPanel: <TData>(props: React.PropsWithChildren<FilterPanelProps<TData>>) => JSX.Element = <TData,>(
```
Comment (odinr) [inline packages/filter/src/components/panel/FilterPanel.tsx:75]: Why is iths file in this PR

File: packages/filter/src/components/panel/FilterPanelProvider.tsx [hunk 14]
```diff
@@ -13,7 +13,7 @@ export type FilterPanelProviderContext = {
 };
 
 const context = createContext<FilterPanelProviderContext | null>(null);
-const { Provider, Consumer } = context;
+const { Provider, Consumer } = context as { Provider: any; Consumer: any };
```
Comment (odinr) [inline packages/filter/src/components/panel/FilterPanelProvider.tsx:16]: Why is iths file in this PR

File: packages/filter/src/context.ts [hunk 15]
```diff
@@ -8,6 +8,6 @@ export const useFilterContext = <
   TData extends Record<string, any> = Record<string, any>,
 >(): FilterContext<TSelection, TData> => useContext(context) as FilterContext<TSelection, TData>;
 
-export const { Provider, Consumer } = context;
+export const { Provider, Consumer } = context as { Provider: any; Consumer: any };
```
Comment (odinr) [inline packages/filter/src/context.ts:11]: Why is iths file in this PR

File: packages/filter/src/options/FilterOptionProvider.tsx [hunk 16]
```diff
@@ -24,7 +24,9 @@ export type FilterComponentProviderProps<TData, TOption extends FilterOption, TV
   initial?: Set<TValue>;
 };
 
-export const FilterOptionProvider = <TData extends Record<string, any>, TOption extends FilterOption, TValue>(
+export const FilterOptionProvider: <TData extends Record<string, any>, TOption extends FilterOption, TValue>(
```
Comment (odinr) [inline packages/filter/src/options/FilterOptionProvider.tsx:27]: Why is iths file in this PR

File: packages/filter/src/options/context.ts [hunk 17]
```diff
@@ -4,7 +4,8 @@ import type { FilterOption, FilterOptionContext } from './types';
 
 // @ts-ignore
 const context = createContext<FilterOptionContext>({});
-export const { Provider, Consumer } = context;
+
```
Comment (odinr) [inline packages/filter/src/options/context.ts:7]: Why is iths file in this PR

File: packages/hanging-garden/src/components/PopoverContainer/index.tsx [hunk 18]
```diff
@@ -18,7 +18,11 @@ const useStyles = makeStyles(() =>
   }),
 );
 
-const PopoverContainer = forwardRef((props: { children: ReactNode }, ref: ForwardedRef<HTMLDivElement>) => {
+const PopoverContainer: React.ForwardRefExoticComponent<
```
Comment (odinr) [inline packages/hanging-garden/src/components/PopoverContainer/index.tsx:21]: Why is iths file in this PR

File: packages/hanging-garden/src/renderHooks/useHangingGardenContext.tsx [hunk 19]
```diff
@@ -40,7 +40,9 @@ export interface IHangingGardenContext {
   padding: number;
 }
 
-const HangingGardenContext = createContext<IHangingGardenContext>({} as IHangingGardenContext);
+const HangingGardenContext: React.Context<IHangingGardenContext> = createContext<IHangingGardenContext>(
```
Comment (odinr) [inline packages/hanging-garden/src/renderHooks/useHangingGardenContext.tsx:43]: Why is iths file in this PR

File: packages/side-sheet/package.json [hunk 20]
```diff
@@ -40,12 +40,12 @@
...
     "@equinor/eds-icons": "^0.19.1",
     "@equinor/eds-tokens": "^0.9.0",
-    "@types/react": "^17.0.11",
+    "@types/react": "^17.0.2",
     "@types/styled-components": "^5.1.26",
     "react": "^17.0.2",
     "react-dom": "^17.0.2",
     "styled-components": "^5.3.9",
-    "typescript": "^4.9.3"
+    "typescript": "^4.4.4"
```
Comment (odinr) [inline packages/side-sheet/package.json:48]: why is typescript downgraded?

**Review by odinr — CHANGES_REQUESTED** [review]
this PR contains 49 commits, maybe rebase since this is a major change
this PR includes changes to files which are not affected by the move to PNPM
some of the workflows would not run, or ruin released packages

**Review by odinr — DISMISSED** [review]

**Pull request merged**